packages/block-editor/src/components/global-styles/border-panel.js

```diff
@@ -50,10 +50,16 @@ function applyFallbackStyle( border ) {
 		return border;
 	}
 
-	if ( ! border.style && ( border.color || border.width ) ) {
+	const hasColorOrWidth = border.color || border.width;
+
+	if ( ! border.style && hasColorOrWidth ) {
 		return { ...border, style: 'solid' };
 	}
 
+	if ( border.style && ! hasColorOrWidth ) {
```

Comment:
Is it possible that a user would want to set a border style and nothing else? Perhaps the border width/color is coming from a theme CSS or other external source?

As it is, selecting only a border style for a block doesn't add that style to the attributes (so I can't save)

Reply:
Thanks for flagging this. Moving the application of the border style fallback to the global styles panel restores the ability to set a style only for the individual block where that makes sense.